tests/core/pyspec/eth2spec/test/deneb/block_processing/test_process_voluntary_exit.py

```diff
@@ -37,9 +38,48 @@ def test_voluntary_exit_with_previous_fork_version_not_is_before_fork_epoch(spec
     """
     assert state.fork.previous_version != state.fork.current_version
 
-    yield from run_voluntary_exit_processing_test(
-        spec,
-        state,
-        fork_version=state.fork.previous_version,
-        is_before_fork_epoch=False,
-    )
+    if spec.fork == DENEB:
+        assert state.fork.previous_version == spec.config.CAPELLA_FORK_VERSION
+        yield from run_voluntary_exit_processing_test(
+            spec,
+            state,
+            fork_version=state.fork.previous_version,
+            is_before_fork_epoch=False,
+        )
+    else:
+        assert state.fork.previous_version != spec.config.CAPELLA_FORK_VERSION
+        yield from run_voluntary_exit_processing_test(
+            spec,
+            state,
+            fork_version=state.fork.previous_version,
+            is_before_fork_epoch=False,
+            valid=False,
+        )
+
+
+@with_deneb_and_later
+@spec_state_test
+@always_bls
+def test_voluntary_exit_with_previous_fork_version_is_before_fork_epoch(spec, state):
+    """
+    Since Deneb, the VoluntaryExit domain is fixed to `CAPELLA_FORK_VERSION`
```

Comment:
```suggestion
    Since Deneb, the VoluntaryExit domain is fixed to `CAPELLA_FORK_VERSION`

    Note: This test is valid for ``spec.fork == DENEB`` and invalid for subsequent forks
```